frontend/src/components/UserDetails.vue

```diff
@@ -173,13 +185,45 @@ export default {
           if (this.user.attributes.org_details) {
             this.user.attributes.org_details = formatOrganization(this.user.attributes.org_details);
           }
-
           this.$store.commit("user/setUser", this.user);
         })
         .catch(e => {
           console.log(e);
         });
     },
+    loadUserRoles: function(){    
+      this.user.allRoles = [];
+      /*
+      No API call exists to load all roles for a user, so we need to first query
+      for a list of clients, then load the roles for each client one by one
+      */     
+      clients.get().then(allClients => {        
+        allClients.data.forEach(client => {
+            UsersRepository.getUserEffectiveClientRoles(this.userId,client.id).then(clientRoles => {
+              if (clientRoles.data.length>0){
+                let roleCollection = {};
+                roleCollection.clientName = client.name;
+                roleCollection.effectiveRoles = clientRoles.data;
+                this.user.allRoles.push(roleCollection);
+                //Sort results by client name
+                this.user.allRoles.sort(function(a, b) {
```

Comment:
I think this is running N sorts, one for each client, when you could sort once at the end.